File: Run-Length Encoding/O(n)_time_and_space.js
function runLengthEncoding(string) {
  // Write your code here.
  ret = [];
  count = 0;
  word = string[0];
  for (let char of string) {
    if (count === 9 || char !== word) {
      ret.push(count.toString());
      ret.push(word);
      word = char;
      count = 0;
    }
    count += 1;
  }
  ret.push(count.toString());
  ret.push(word);

  return ret.join("");
}

File: Run-Length Encoding/O(n)_time_and_space.py
def runLengthEncoding(string):
    # Write your code here.
    ret=[]
    count=0
    word=string[0]
    for char in string:
        if count ==9 or word !=char:
            ret.append(str(count))
            ret.append(word)
            word=char
            count=0
        count+=1
    ret.append(str(count))
    ret.append(word)
    return "".join(ret)
